src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs

```diff
@@ -140,11 +142,27 @@ private set
             _serversPing = new CustomizedPingTimer(Logger, Constants.CustomizedPingTimer.Servers, WriteServersPingAsync, Constants.Periods.DefaultServersPingInterval, Constants.Periods.DefaultServersPingInterval);
         }
 
-        public Task StartAsync() => Task.WhenAll(FixedServiceConnections.Select(c => StartCoreAsync(c)));
+        // todo: maybe create a new container-level ServiceConnectionContainerScope here, to reduce duplicate events on client-level.
+        public async Task StartAsync()
+        {
+            var context = new ServiceDiagnosticLogsContext { IsEnableMessageLog = false };
+            using (new ServiceConnectionContainerScope(context))

```

Comment:
I would much rather prefer explicit (if possible) wiring than creating async locals here and there. 
Although this context is not carrying references to any other objects so the overhead and the risk of leaking is likely small.